src/ripple/ledger/impl/PaymentSandbox.cpp

```diff
@@ -180,6 +213,19 @@ PaymentSandbox::balanceHook (AccountID const& account,
     return adjustedAmt;
 }
 
+std::uint32_t
+PaymentSandbox::ownerCountHook (AccountID const& account,
+    std::uint32_t count) const
+{
+    std::uint32_t result = count;
+    for (auto curSB = this; curSB; curSB = curSB->ps_)
+    {
+        if (auto adj = curSB->tab_.ownerCount (account))
+            result = std::max (result, *adj);
```

Comment:
Steps in a payment path are not allowed to use assets acquired during that payment. We already track credits and debits. We were not tracking owner counts. If an owner count decreases, that decreases the reserve and thus increases the assets available. We shouldn't allow that.

`ownerCountHook` is similar to `balanceHook`. It takes the current value and adjusts it so assets acquired during the payment are not used. Since payments never increase owner counts, taking the max is the same as "ignoring the credits and subtracting the debits".

So this isn't just "being conservative". If we don't do this payments can act in strange ways.

I also left a comment on `ownerCountHook`